### PR TITLE
Improved building the engine source for Android on Win32 machines

### PR DESCRIPTION
**Pull request by JCash**
We've streamlined the build process for Android a bit when building the source on a Windows machine.
It will pick up the Android Studio installation from the default location (`%LOCALAPPDATA%/Android/Sdk`) or `ANDROID_HOME`.
This will be helpful for new contributors.

Note that we still currently have the requirement of NDK 25 being installed.

## PR checklist

* [ ] Code
	* [ ] Add engine and/or editor unit tests.
	* [ ] New and changed code follows the overall code style of existing code
	* [ ] Add comments where needed
* [ ] Documentation
	* [ ] Make sure that API documentation is updated in code comments
	* [ ] Make sure that manuals are updated (in github.com/defold/doc)
* [ ] Prepare pull request and affected issue for automatic release notes generator
	* [ ] Pull request - Write a message that explains what this pull request does. What was the problem? How was it solved? What are the changes to APIs or the new APIs introduced? This message will be used in the generated release notes. Make sure it is well written and understandable for a user of Defold.
	* [ ] Pull request - Write a pull request title that in a sentence summarises what the pull request does. Do not include "Issue-1234 ..." in the title. This text will be used in the generated release notes.
	* [ ] Pull request - Link the pull request to the issue(s) it is closing. Use on of the [approved closing keywords](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue).
	* [ ] Affected issue - Assign the issue to a project. Do not assign the pull request to a project if there is an issue which the pull request closes.
	* [ ] Affected issue - Assign the "breaking change" label to the issue if introducing a breaking change.
	* [ ] Affected issue - Assign the "skip release notes" is the issue should not be included in the generated release notes.

----------

Example of a well written PR description:

1. Start with the user facing changes. This will end up in the release notes.
1. Add one of the GitHub approved closing keywords
1. Optionally also add the technical changes made. This is information that might help the reviewer. It will not show up in the release notes. Technical changes are identified by a line starting with one of these:
   1. `### Technical changes` 
   1. `Technical changes:`
   2. `Technical notes:`

```
There was a anomaly in the carbon chroniton propeller, introduced in version 8.10.2. This fix will make sure to reset the phaser collector on application startup.

Fixes #1234

### Technical changes
* Pay special attention to line 23 of phaser_collector.clj as it contains some interesting optimizations
* The propeller code was not taking into account a negative phase.
```

### CODE REVIEW

File: build_tools/sdk.py [hunk 1]
```diff
@@ -199,6 +226,9 @@ def get_android_local_sdk_path(verbose=False):
         # on macOS, it doesn't set an environment variable
         if sys.platform == 'darwin':
             path = os.path.expanduser('~/Library/android/sdk')
+        elif sys.platform == 'win32':
+            path = os.path.expandvars('${LOCALAPPDATA}/Android/Sdk')
+        path = os.path.normpath(path)
```
Comment (JCash) [inline build_tools/sdk.py:231]: Not sure where linux stores it, but they can use ANDROID_HOME to set the root.

File: build_tools/sdk.py [hunk 2]
```diff
@@ -228,6 +258,15 @@ def get_android_local_build_tools_path(platform):
 def get_android_local_sdk_version(platform):
     return os.path.basename(get_android_local_build_tools_path(platform))
 
+def get_android_local_jar_path(verbose=False):
```
Comment (JCash) [inline build_tools/sdk.py:261]: The android.jar is a central piece of the android sdk.

File: build_tools/sdk.py [hunk 3]
```diff
@@ -596,6 +636,7 @@ def _get_local_sdk_info(platform, verbose=False):
         info['version'] = get_android_local_sdk_version(platform)
         info['sdk'] = get_android_local_sdk_path(verbose)
         info['ndk'] = get_android_local_ndk_path(platform, verbose)
+        info['jar'] = get_android_local_jar_path(platform, verbose)
```
Comment (JCash) [inline build_tools/sdk.py:639]: Now we also return the jar file location

File: build_tools/waf_dynamo.py [hunk 4]
```diff
@@ -1027,8 +1037,10 @@ def android_package(task):
     except BuildUtilityException as ex:
         task.fatal(ex.msg)
 
+    sdk_info = sdk.get_sdk_info(SDK_ROOT, build_util.get_target_platform())
+    android_jar = sdk_info['jar'] # android.jar from the sdk
```
Comment (JCash) [inline build_tools/waf_dynamo.py:1041]: Either local sdk or the defold prepackaged path.

File: build_tools/waf_dynamo.py [hunk 5]
```diff
@@ -1648,29 +1660,40 @@ def detect(conf):
...
             conf.fatal("Path does not exist: %s" % bintools)
 
-        conf.env['CC']       = '%s/%s' % (bintools, clang_name)
-        conf.env['CXX']      = '%s/%s++' % (bintools, clang_name)
-        conf.env['LINK_CXX'] = '%s/%s++' % (bintools, clang_name)
-        conf.env['CPP']      = '%s/%s -E' % (bintools, clang_name)
-        conf.env['AR']       = '%s/%s-ar' % (bintools, tool_name)
-        conf.env['RANLIB']   = '%s/%s-ranlib' % (bintools, tool_name)
-        conf.env['LD']       = '%s/lld' % (bintools)
+        conf.env['ANDROID_JAR'] = sdkinfo['jar']
```
Comment (JCash) [inline build_tools/waf_dynamo.py:1684]: Our internal wscripts now use this variable.

**Review by AGulev — APPROVED** [review]
it looks good to me

**Pull request merged**